RequestManagementSystem/Agent/RequestExecutingAgent.py

```diff
@@ -89,263 +94,269 @@ class RequestExecutingAgent( AgentModule ):
+          self.log.error("execute:", "%s" % getRequests["Message"])
           break
         if not getRequests["Value"]:
-          self.log.info( "execute: no more 'Waiting' requests to process" )
+          self.log.info("execute: no more 'Waiting' requests to process")
           break
         for rId in getRequests["Value"]["Failed"]:
-          self.log.error( "execute: %s" % getRequests["Value"]["Failed"][rId] )
+          self.log.error("execute:", "%s" % getRequests["Value"]["Failed"][rId])
 
         requestsToExecute = getRequests["Value"]["Successful"].values()
 
-      self.log.info( "execute: will execute %s requests " % len( requestsToExecute ) )
+      self.log.info("execute: will execute requests ", "%s" % len(requestsToExecute))
 
       for request in requestsToExecute:
         # # set task id
         taskID = request.RequestID
 
-        self.log.info( "processPool tasks idle = %s working = %s" % ( self.processPool().getNumIdleProcesses(),
-                                                                      self.processPool().getNumWorkingProcesses() ) )
+        self.log.info("processPool status", "tasks idle = %s working = %s" %
+                      (self.processPool().getNumIdleProcesses(), self.processPool().getNumWorkingProcesses()))
 
         looping = 0
         while True:
           if not self.processPool().getFreeSlots():
             if not looping:
-              self.log.info( "No free slots available in processPool, will wait %d seconds to proceed" % self.__poolSleep )
-            time.sleep( self.__poolSleep )
+              self.log.info(
+                  "No free slots available in processPool",
+                  "will wait %d seconds to proceed" %
+                  self.__poolSleep)
+            time.sleep(self.__poolSleep)
             looping += 1
           else:
             if looping:
-              self.log.info( "Free slot found after %d seconds" % looping * self.__poolSleep )
+              self.log.info("Free slot found", "after %d seconds" % looping * self.__poolSleep)
             looping = 0
             # # save current request in cache
-            res = self.cacheRequest( request )
+            res = self.cacheRequest(request)
             if not res['OK']:
-              if cmpError( res, errno.EALREADY ):
+              if cmpError(res, errno.EALREADY):
                 # The request is already in the cache, skip it. break out of the while loop to get next request
                 break
               # There are too many requests in the cache, commit suicide
-              self.log.error( res['Message'], '(%d requests): put back all requests and exit cycle' % len( self.__requestCache ) )
+              self.log.error(
+                  "Too many requests in cache",
+                  '(%d requests): put back all requests and exit cycle. Error %s' %
+                  len(
+                      self.__requestCache,
```

Comment:
You need to close the `len(` parenthesis here, and add parenthesis after `%` around the tuple

```python
'.... Error %s' % (len(self.__...), res['Message']))
```

Reply:
Oups indeed ! Wonder why this was not detected though...

Reply:
Fixed, thanks